### PR TITLE
Remove qiskit.tools module

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit removes all the functionality in `qiskit.tools` with the exception of `parallel_map` which has been migrated to `qiskit.utils`. This PR does not cover the removal of `qiskit.tools.jupyter` or `qiskit.tools.monitor` which are covered by #11513.

### Details and comments

This should probably merge after #11513, there shouldn't be a merge conflict (if I did it correctly), but the removal is incomplete without #11513 and I expect there will be some test failures related to that.

### CODE REVIEW

File: qiskit/utils/__init__.py [hunk 1]
```diff
@@ -30,6 +30,13 @@
 .. autofunction:: wrap_method
 
 
+Parallel Routines
+-----------------
+A helper function for calling a custom function with python
+``ProcessPoolExecutor``. Tasks can be executed in parallel using this function.
```
Comment (jakelishman) [inline qiskit/utils/__init__.py]: We have intersphinx set up to target the stdlib, so if you really want you can do
```suggestion
:class:`~concurrent.futures.ProcessPoolExecutor`. Tasks can be executed in parallel using this function.
```
not necessary, though

File: qiskit/utils/parallel.py [hunk 2]
```diff
@@ -151,13 +150,6 @@ def func(_):
     if len(values) == 1:
         return [task(values[0], *task_args, **task_kwargs)]
 
-    Publisher().publish("terra.parallel.start", len(values))
-    nfinished = [0]
-
-    def _callback(_):
-        nfinished[0] += 1
-        Publisher().publish("terra.parallel.done", nfinished[0])
-
```
Comment (jakelishman) [inline qiskit/utils/parallel.py:160]: What was the point of this, historically?

**Review by jakelishman — COMMENTED** [review]
This seems pleasantly small and an easy clean removal.  The move of `parallel_map` is a shade unfortunate, but we talked about that on #11514.

**Review by jakelishman — DISMISSED** [review]
(meant to click approve on that review)

File: qiskit/utils/__init__.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/utils/__init__.py]: Oh, that's a good idea. I'll do that

**Review by jakelishman — APPROVED** [review]

File: qiskit/utils/parallel.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/utils/parallel.py:160]: It was originally used to have a progress bar for things like `parallel_map`. Overall there was a goal to have a decoupled event bus that different components could subscribe to for async processing. But it never was really used for anything more than progress bars.

Comment (jakelishman) [reply qiskit/utils/parallel.py:160]: Ah, good to know, thanks.  It was one of the major differences from the version that lived in QuTiP, but I'd never actually looked into it to find out what it was for.

**Pull request merged**